src/main/java/com/openlattice/organizations/HazelcastOrganizationService.java

```diff
@@ -320,25 +323,37 @@ private void addMembers( AclKey orgAclKey, Set<Principal> members ) {
     }
 
     public void removeMembers( UUID organizationId, Set<Principal> members ) {
+        final var users = members.stream()
+                .filter( m -> m.getType().equals( PrincipalType.USER ) )
+                .collect( Collectors.toList() );
+        final var securablePrincipals = securePrincipalsManager.getSecurablePrincipals( users );
+
         removeRolesFromMembers(
-                getRolesInFull( organizationId ).stream().map( Role::getAclKey ),
-                members
-                        .stream()
-                        .filter( m -> m.getType().equals( PrincipalType.USER ) )
-                        .map( securePrincipalsManager::lookup ) );
+                getRoles( organizationId ).stream().map( Role::getAclKey ).collect( Collectors.toSet() ),
+                securablePrincipals
+        );
         membersOf.executeOnKey( organizationId, new OrganizationMemberRemover( members ) );
-
         final AclKey orgAclKey = new AclKey( organizationId );
-        members.stream().filter( PrincipalType.USER::equals )
-                .map( securePrincipalsManager::lookup )
-                .forEach( target -> securePrincipalsManager.removePrincipalFromPrincipal( orgAclKey, target ) );
+        removeOrganizationFromMembers( orgAclKey, securablePrincipals );
+
+        eventBus.post(
+                new MembersRemovedFromOrganizationEvent(
+                        organizationId,
+                        new SecurablePrincipalList( securablePrincipals )
+                )
+        );
+    }
 
-        eventBus.post( new MembersRemovedFromOrganizationEvent( organizationId, new PrincipalSet( members ) ) );
+    private void removeRolesFromMembers( Set<AclKey> roles, Collection<SecurablePrincipal> members ) {
+        members.forEach( securablePrincipal ->
```

Comment:
I think this can still be bulkier -- this will still be calling `principalTrees.executeOnKey` for every member, but you can just pass all the member aclKeys through and call `principalTrees.executeOnKeys` for them all at once

Reply:
Yeah, right, thanks!